qutip/tests/test_correlation.py

```diff
@@ -251,3 +253,64 @@ def test_hamiltonian_order_unimportant():
     backwards = qutip.correlation_2op_2t(H[::-1], start, times, times, [sp],
                                          sp.dag(), sp)
     np.testing.assert_allclose(forwards, backwards, atol=1e-6)
+
+
+# Paramters for the test
+solver_list = ['me', 'es']
+start_list = [ 'coherent_dm', 'fock', None]
```

Comment:
I'm not sure you need to test the case with `None` - since you know your Hamiltonian and your collapse operator, your steady state will always just be `|0><0|`, right?  It might be better to parametrise over 2 different values of the Fock state (say 1 and 5), and parametrise over `dm=[True, False]` (tied to the solver in use) instead?

Reply:
I do actually think that the `None` case should be tested. Accepting `None` as the initial state is supposed to be a "feature" in which case we should test that it behaves properly.

Apart from `None`, the possible initial states could be a ket or a dm. Both cases are already covered with `coherent_dm` and `fock` which were chosen to be that way just because they already appeared in the file. I would not change these cases since I believe they already test thoroughly enough `correlation_2op_1t`, although if you still think it is necessary I have no problem with changing them.

Reply:
You're right that we need to test the steady-state functionality of `correlation_2op_1t`.  I'd thought it was tested somewhere else in the file already.  I'm fine with the test cases as-is.